solt/transforms/_transforms.py

```diff
@@ -1212,6 +1214,55 @@ def _apply_img(self, img: np.ndarray, settings: dict):
         return cv2.LUT(img, self.state_dict["LUT"])
 
 
+class IntensityRemap(ImageTransform):
+    """Performs random intensity remapping.
+
+    Parameters
+    ----------
+    p : float
+        Probability of applying this transform,
+    kernel_size: int
+        Size of medial filter kernel used during the generation of intensity mapping.
+        Higher value yield more monotonic mapping.
+    data_indices : tuple or None
+        Indices of the images within the data container to which this transform needs to be applied.
+        Every element within the tuple must be integer numbers.
+        If None, then the transform will be applied to all the images withing the DataContainer.
+
+    References
+    ----------
+    .. [1] Hesse, L. S., Kuling, G., Veta, M., & Martel, A. L. (2019).
+       Intensity augmentation for domain transfer of whole breast
+       segmentation in MRI. https://arxiv.org/abs/1909.02642
+    """
+
+    serializable_name = "intensity_remap"
+    """How the class should be stored in the registry"""
+
+    def __init__(self, kernel_size=9, data_indices=None, p=0.5):
+        super(IntensityRemap, self).__init__(p=p, data_indices=data_indices)
+        self.kernel_size = kernel_size
+
+    def sample_transform(self, data):
+        m = random.sample(range(256), k=256)
+        m = scipy.signal.medfilt(m, kernel_size=self.kernel_size)
+        m = m + np.linspace(0, 255, 256)
+
+        m = m - min(m)
+        m = m / max(m) * 255
+        m = np.floor(m).astype(np.uint8)
+
+        self.state_dict = {"LUT": m}
+
+    @img_shape_checker
+    def _apply_img(self, img: np.ndarray, settings: dict):
+        if img.dtype != np.uint8:
+            raise ValueError("IntensityRemap supports uint8 ndarrays only")
+        if img.ndim == 3 and img.shape[-1] != 1:
```

Comment:
Why to limit to grayscale? I believe that it is more generic to apply the transform per channel.

Reply:
I am not sure what would be the equivalent transform for multichannel images and I have not seen any publications on assessing the efficacy of such transform in, e.g. RGB case. Just mapping the values, considering the channel dimension as spatial, would not do. The correct analogue would probably imply conditional sampling from the HSV space or something like that.

Reply:
Still, for compatibility issues with th pretrained backbones, some people often read the grayscale as RGB / BGR. I suggest to add a support for this..